aws/aws-toolkit-jetbrains
# Add support to run Lambda function locally

## Types of changes

## Description
- Add ability to run labda in Rider through SAM CLI invocations;
- Split function to check if lambda is valid with check for PSI element for handler string. Run configuration is validated over same function call that get PSI element. For Rider we have to make this call to R# backend. Run configuration validation mechanism is synchronious, so, we have a separate quick method that check whether method exist and other one to get method information found on a backend based on handler string
- Fix issue related to indexing DLL files by specifying the Runtime supported languages
- Add Rider frontend tests to verify Lambda Handler

## Motivation and Context
Add ability to run Lambda locally through Rider IDE to make sure it runs and gives a correct results before deploy to production.

## Related Issue(s)
None.

## Testing
- Run Lambda locally through run configuration and check it is executed.
- Check configuration validation
- Add support for Rider Test Framework and add dependent tests to verify logic that look for a method from a handler string
- Verified on Mac OS X

## Screenshots (if appropriate)
<img width="949" alt="AWS_Rider_RunLambdaLocal" src="https://user-images.githubusercontent.com/6064345/59267783-bb0ee300-8c53-11e9-878f-6e4b75547762.png">

## Checklist

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.
